### ISSUE TITLE
Docker for Desktop Preview M1 - hangs on startup if macOS Firewall is set to "Block all incoming connections"

### ISSUE THREAD

**Issue by tkopczuk** [docker/for-mac#5140]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: BF494B06-5BEA-4251-B5F2-8DC099F9D039/20201216152920

### Expected behavior

It should start and run correctly, whilst disallowing inbound connections made from external networks to any container bound to `0.0.0.0`.

That was the Intel version behavior.

### Actual behavior

It hangs on "Starting".

### Information

  - macOS Version: macOS 11.1 (20C69)

I believe it's important to explore potential workarounds, as it's, in my experience, a common MDM-enforced setting for many software development companies.

In our case it's aimed to mitigate the risk of someone accidentally binding eg. a development PostgreSQL or a pre-release software to the `0.0.0.0` interface on a shared WiFi network.

It's common for databases used for development to either use a default password or not enforce authentication at all.

Additional context for the importance of blocking all connections instead of a more lenient per-app setting:  
[Docker not blocked by macOS firewall #729](https://github.com/docker/for-mac/issues/729)

### Diagnostic logs

```
Docker for Mac: Preview 5

INFO[0002] connecting to tcp+connect://192.168.64.1:1999...
INFO[0033] cannot set up multiplexer, will retry in 2s: dial tcp 192.168.64.1:1999: connect: connection timed out
INFO[0035] connecting to tcp+connect://192.168.64.1:1999...
INFO[0067] cannot set up multiplexer, will retry in 2s: dial tcp 192.168.64.1:1999: connect: connection timed out
INFO[0069] connecting to tcp+connect://192.168.64.1:1999...
INFO[0101] cannot set up multiplexer, will retry in 2s: dial tcp 192.168.64.1:1999: connect: connection timed out
INFO[0103] connecting to tcp+connect://192.168.64.1:1999...
INFO[0135] cannot set up multiplexer, will retry in 2s: dial tcp 192.168.64.1:1999: connect: connection timed out
INFO[0137] connecting to tcp+connect://192.168.64.1:1999...
INFO[0168] cannot set up multiplexer, will retry in 2s: dial tcp 192.168.64.1:1999: connect: connection timed out
```

### Steps to reproduce the behavior

  1. Open System Preferences.app
  2. Navigate to Security & Privacy -> Firewall.
  3. Turn the Firewall On.
  4. Click "Firewall Options..."
  5. Check "Block all incoming connections"
  6. Start Docker for Mac.

**Comment by xiazemin** [comment]
/Library/preferences/SystemConfiguration/com.apple.vmnet.plist

        <key>Shared_Net_Address</key>
        <string>192.168.65.1</string>
        <key>Shared_Net_Mask</key>
        <string>255.255.255.0</string>

**Comment by andymac4182** [comment]
Any ideas how to resolve this? We have an MDM that is required to enforce the Block all incoming connections.

**Comment by macmladen** [comment]
@xiazemin Not sure what you meant to say but that piece is already in the plist?

**Comment by andymac4182** [comment]
I found a way to ignore the MDM for a bit `sudo /usr/libexec/ApplicationFirewall/socketfilterfw --setblockall off` thanks to someone in another issue.

**Comment by andymac4182** [comment]
Confirmed this is still an issue on the released version

**Comment by andymac4182** [comment]
Sorry this is working now for me with the released version. Version 3.3.1
